linkerd/buffer/src/error.rs

```diff
@@ -36,3 +40,19 @@ impl std::error::Error for ServiceError {
         Some(&**self.0.as_ref())
     }
 }
+
+// === impl IdleError ===
+
+impl<'a> std::fmt::Display for IdleError {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let secs = self.0.as_secs();
+        let subsec_ms = self.0.subsec_nanos() as f64 / 1_000_000f64;
+        if secs == 0 {
+            write!(fmt, "Service idled out after {}ms", subsec_ms)
+        } else {
+            write!(fmt, "Service idled out after {}s", secs as f64 + subsec_ms)
```

Comment:
If I'm reading this correctly, this looks like it will add `subsec_ms` to `secs` as though they were a number of seconds? Don't we need to divide them so they are only added to the fractional part of `secs`?

Reply:
(I think this also applies to the version of this change that merged to master, I just didn't notice it earlier)